pimd/pim_upstream.c

```diff
@@ -1830,30 +1830,31 @@ bool pim_upstream_equal(const void *arg1, const void *arg2)
  */
 static bool pim_upstream_kat_start_ok(struct pim_upstream *up)
 {
-	struct pim_instance *pim = up->channel_oil->pim;
+	struct channel_oil *c_oil = up->channel_oil;
+	struct interface *ifp = up->rpf.source_nexthop.interface;
+	struct pim_interface *pim_ifp;
+
+	/* "iif == RPF_interface(S)" check is not easy to do as the info
+	 * we get from the kernel/ASIC is really a "lookup/key hit".
+	 * So we will do an approximate check here to avoid starting KAT
+	 * because of (S,G,rpt) forwarding on a non-LHR.
+	 */
+	if (!ifp)
+		return false;
+
+	pim_ifp = ifp->info;
```

Comment:
we need to check for null on pim_ifp. if pim disabled on the rpf i/f, pim_ifp will be null and this may lead to crash.

Reply:
Can an interface even be added as RPF if it is not pim-enabled?

Reply:
Can an interface even be added as RPF if it is not pim-enabled?

Reply:
wait?  I'm pretty sure we create a pim_ifp no matter what.

Reply:
@donaldsharp  We dont create pim_ifp, if pim is not enabled.
@AnuradhaKaruppiah PIm not enabled interfaces are stored in pnc because we may not get the nht notification when pim is enabled on those interfaces.

Reply:
An interface that is not pim-enabled may be stored in the PNC  but it will not be set as upstrea->rpf.source_nexthop.interface.

Reference: 
pim_ecmp_nexthop_lookup()
{
>>>>>>>>>>>>>>>>> SNIP >>>>>>>>>>>>>>>>>>
        if (!ifp->info) {
            if (PIM_DEBUG_PIM_NHT)
                zlog_debug(
                    "%s: multicast not enabled on input interface %s(%s) (ifindex=%d, RPF for source %s)",
                    __PRETTY_FUNCTION__, ifp->name,
                    pim->vrf->name, first_ifindex,
                    addr_str);
            if (i == mod_val)
                mod_val++;
            i++;
            continue;
        }
>>>>>>>>>>>>>>>>>>> SNIP >>>>>>>>>>>>>>>>>>>
}